domain/schema/controller.go

```diff
@@ -532,3 +533,46 @@ INSERT INTO autocert_cache_encoding VALUES
     (0, 'x509');    -- Only x509 certs encoding supported today.
 `)
 }
+
+func userSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE user (
+    uuid            TEXT PRIMARY KEY,
+    name            TEXT NOT NULL,
+    display_name    TEXT,
+    removed         BOOLEAN NOT NULL,
+    created_by_uuid TEXT,
+    created_at      TIMESTAMP NOT NULL,
+    CONSTRAINT      fk_user_created_by_user
+        FOREIGN KEY (created_by_uuid)
+    REFERENCES      user(uuid)
+);
+
+CREATE UNIQUE INDEX idx_singleton_active_user ON user (name) WHERE deleted IS FALSE;
+
+CREATE TABLE user_last_login (
```

Comment:
This appears to have been a separate collection in Mongo in order to live outside the transaction operation workflow.

We don't need to mirror for Dqlite; just add `last_login` to the user table.

Reply:
Moved to the user table.

Reply:
With the current ddl as we have it we have what I think is a cool concept in that we can have the notion of a user/owner in Juju for the purpose of ownership or user interfaces with the user necessarily having to authentication information.

My example would be let's say that you have a `jujud` snap where you can bootstrap a controller without a client we still need a owner of the model and everything else but not necessarily something that can or is logged in.

In my opinion the design we should be going for in the DDL is identity, authn, authz. We conflate identity and authn together by moving last_login onto the same table.

Reply:
Thx, know I hope the issue is resolved.